nixos/modules/services/databases/postgresql.nix

```
@@ -391,12 +391,11 @@ in
       };
 
       extraPlugins = mkOption {
-        type = types.listOf types.path;
-        default = [];
-        example = literalExpression "with pkgs.postgresql_15.pkgs; [ postgis pg_repack ]";
+        type = with types; functionTo (listOf path);
```

```suggestion
        type = with types; coercedTo (listOf path) (path: _ignorePg: path) (functionTo (listOf path));
```

I believe this would be backwards compatible. You could check by temporarily reverting a test and instantiating it.